qt/applications/workbench/workbench/plugins/workspacewidget.py

```diff
@@ -260,6 +260,15 @@ def _do_slice_viewer(self, names):
         parent, flags = get_window_config()
         for ws in self._ads.retrieveWorkspaces(names, unrollGroups=True):
             try:
+                specs = ws.getSpectrumNumbers()
+                y = ws.getYDimension()
+                if y.getName() == "sample":
+                    spec_to_y_dim = {s: y.getX(s) for s in specs}
```

Comment:
Not sure the benefit of using a `dict` here - the keys don't seem to be used.
I would probably do
```suggestion
                    spec_xvals = [ws.getYDimension().getX(ispec) for ispec in range(ws.getNumberHistograms())]
```
Note that `getX` requires the spectrum index not spectrum number.
Then you don't need to access `.values()` to check it's sorted, you could do
```
if sorted(spec_xvals) != spec_xvals:
```